lib/spack/spack/cmd/install.py

```diff
@@ -35,12 +35,15 @@
 
 def setup_parser(subparser):
     subparser.add_argument(
-        '-i', '--ignore-dependencies', action='store_true', dest='ignore_deps',
-        help="Do not try to install dependencies of requested packages.")
-    subparser.add_argument(
-        '-d', '--dependencies-only', action='store_true', dest='deps_only',
-        help='Install dependencies of this package, ' +
-        'but not the package itself.')
+        '--only',
+        default='package:dependencies',
+        dest='things_to_install',
+        choices=['package', 'dependencies', 'package:dependencies'],
```

Comment:
Should we also let the user do dependencies:packages here?  I proposed the mini-parser to allow this flexibility in order; but for just 2 options, just adding them this way might be another approach.  Probably not worth changing, since 'package:dependencies' is already the default, users will rarely use it.